independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ListExtensionsCommandHandler.java

```diff
@@ -49,20 +52,27 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
         } catch (IOException e) {
             throw new QuarkusCommandException("Failed to determine the list of installed extensions", e);
         }
-
-        Stream<Extension> platformExtensionsStream = invocation.getPlatformDescriptor().getExtensions().stream();
-        platformExtensionsStream = platformExtensionsStream.filter(this::filterUnlisted);
-        if (search != null && !"*".equalsIgnoreCase(search)) {
-            final Pattern searchPattern = Pattern.compile(".*" + search + ".*", Pattern.CASE_INSENSITIVE);
-            platformExtensionsStream = platformExtensionsStream.filter(e -> filterBySearch(searchPattern, e));
+        final Collection<Extension> platformExtensions;
+        if (extensionRegistry != null) {
+            String quarkusVersion = invocation.getPlatformDescriptor().getQuarkusVersion();
+            Collection<Extension> result = extensionRegistry.list(quarkusVersion, search);
+            if (result.isEmpty()) {
+                log.warn(
+                        "No extension found with pattern '%s' while searching in the configured extension registries (using Quarkus Core %s).\n"
+                                + "Falling back to existing platform extensions",
+                        search, quarkusVersion);
+                platformExtensions = searchExtensions(invocation, search);
```

Comment:
What is happening here? What is this fallback about?

Reply:
Why are we having two sources of extensions?

Reply:
> What is happening here? What is this fallback about?
> Why are we having two sources of extensions?

The fallback happens to make sure that the search is also performed in the platform BOM that is associated to the project. This will likely happen if the specified registry does not list the Quarkus core version used in the project

Reply:
The fact that they aren't unified is strange to me. Perhaps I'm misunderstood the use-case.